scripts/helper_scripts/new-parsers/src/utils/files.rs

```diff
@@ -0,0 +1,32 @@
+use std::fs::{File, OpenOptions};
+use std::io::{stdin, BufReader, BufWriter, Stdin};
+use std::path::PathBuf;
+
+/// Create a BufReader that reads from StdIn
+pub fn open_sin() -> BufReader<Stdin> {
+    BufReader::new(stdin())
+}
+
+/// Create a BufReader that reads from a file denoted by its PathBuf
+pub fn open_read(pb: &PathBuf) -> BufReader<File> {
+    let file = OpenOptions::new().read(true).open(pb);
+    match file {
+        Ok(f) => BufReader::new(f),
+        Err(e) => {
+            eprintln!("{:?}", e);
+            std::process::exit(1);
+        }
+    }
+}
```

Comment:
If you mean the `error-chain` library, that has been unmaintained for 3+ years and is archived, so I'll look around for something else

Reply:
I can confirm `anyhow` is the way to go for CLI applications. The `?`-syntax is very elegant as well.

Reply:
Aw, I liked error-chain :-( now I should port UMGAP as well... I had a look at anyhow and thiserror. Former looks like the better fit here; thiserror seems better if you want to inspect and recover from the errors (but since these are internal tools, not libraries, you'd not need that).